sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/HiveThriftServer2Listener.scala

```diff
@@ -131,60 +132,81 @@ private[thriftserver] class HiveThriftServer2Listener(
     updateLiveStore(session)
   }
 
-  private def onSessionClosed(e: SparkListenerThriftServerSessionClosed): Unit = {
-    val session = sessionList.get(e.sessionId)
-    session.finishTimestamp = e.finishTime
-    updateStoreWithTriggerEnabled(session)
-    sessionList.remove(e.sessionId)
-  }
+  private def onSessionClosed(e: SparkListenerThriftServerSessionClosed): Unit =
+    Option(sessionList.get(e.sessionId)) match {
+      case None => logWarning(s"onSessionClosed called with unknown session id: ${e.sessionId}")
+      case Some(sessionData) =>
+        val session = sessionData
+        session.finishTimestamp = e.finishTime
+        updateStoreWithTriggerEnabled(session)
+        sessionList.remove(e.sessionId)
+    }
 
-  private def onOperationStart(e: SparkListenerThriftServerOperationStart): Unit = {
-    val info = getOrCreateExecution(
-      e.id,
-      e.statement,
-      e.sessionId,
-      e.startTime,
-      e.userName)
-
-    info.state = ExecutionState.STARTED
-    executionList.put(e.id, info)
-    sessionList.get(e.sessionId).totalExecution += 1
-    executionList.get(e.id).groupId = e.groupId
-    updateLiveStore(executionList.get(e.id))
-    updateLiveStore(sessionList.get(e.sessionId))
-  }
+  private def onOperationStart(e: SparkListenerThriftServerOperationStart): Unit =
+    Option(sessionList.get(e.sessionId)) match {
+      case None => logWarning(s"onOperationStart called with unknown session id: ${e.sessionId}")
```

Comment:
We can keep processing queries even in this case?

Reply:
`sessionList.get(e.sessionId).totalExecution += 1` would result in an NPE if sessionList did not have the sessionId, so it was not possible previously to continue processing.

Reply:
@maropu this code is in a listener that executes on the listener bus. The listener bus can drop events if it becomes to busy.
So it might happen that events are dropped, and id is unavailable. It will not affect query execution.

Reply:
> @maropu this code is in a listener that executes on the listener bus. The listener bus can drop events if it becomes to busy.
So it might happen that events are dropped, and id is unavailable. It will not affect query execution.

Does a server get back to a normal state by just ignoring the case? What I a bit worry about is that users can get the same warning message repeatedly if a session id gets lost (I'm not sure the case can happen in event sequences).

Reply:
@maropu the server does get to a normal state. This is used only for displaying SparkUI, so wouldn't affect the server functioning.
Before this change, a NullPointerException would be thrown on `sessionList.get(e.sessionId).totalExecution += 1`, so I think a WARN message is strictly better than a NPE in the logs.
But good catch that it could be further improved - don't fail on missing session, just log the query with that sessionId, and just don't do `sessionList.get(e.sessionId).totalExecution += 1`

Reply:
Ah, I see. Thanks for the explanation. Looks okay.